packages/Kokkos/src/details/DTK_DetailsTreeConstruction.cpp

```diff
@@ -0,0 +1,294 @@
+#include <details/DTK_DetailsAlgorithms.hpp>
+#include <details/DTK_DetailsTreeConstruction.hpp>
+
+#include <algorithm>
+#include <atomic>
+#include <cassert> // TODO: probably want to use something else but fine for now
+
+namespace DataTransferKit
+{
+namespace Details
+{
+
+// Expands a 10-bit integer into 30 bits
+// by inserting 2 zeros after each bit.
+unsigned int expandBits( unsigned int v )
```

Comment:
Think about templating this function and the one below to be able to use `unsigned long long int`